tests/framework/gitlint_rules.py

```diff
@@ -8,20 +8,70 @@
 # pylint: disable=R0903
 
 
-class SignedOffBy(CommitRule):
-    """Make sure that each commit contains a "Signed-off-by" line."""
+class EndsSigned(CommitRule):
+    """Checks commit message body formatting.
+
+    Makes sure each commit message body ends with
+    1 or more signatures ("Signed-off-by"), followed by
+    0 or more co-authors ("Co-authored-by").
+    """
```

Comment:
All tests pass now (besides platform specific). There were several changes to the original code. I have selected some that seem especially important.